src/uu/base32/src/base_common.rs

```diff
@@ -171,52 +171,29 @@ pub fn get_input(config: &Config) -> UResult<Box<dyn ReadSeek>> {
     }
 }
 
-/// Determines if the input buffer ends with padding ('=') after trimming trailing whitespace.
-fn read_and_has_padding<R: Read>(input: &mut R) -> UResult<(bool, Vec<u8>)> {
-    let mut buf = Vec::new();
-    input
-        .read_to_end(&mut buf)
-        .map_err(|err| USimpleError::new(1, format_read_error(err.kind())))?;
```

Comment:
Hi, I've faced with the same issue: base64 doesn't support streaming -- I wanted to prepare a fix, but then I've seen this PR with potential fix -- but in 5 lines above the issue persists:  whole stdin is read into memory.

I can make a fix on top of this PR